libcudacxx/include/cuda/std/__atomic/functions/cuda_local.h

```diff
@@ -0,0 +1,240 @@
+#  pragma system_header
+#endif // no system header
+
+#include <cuda/std/__atomic/types/common.h>
+#include <cuda/std/cstdint>
+
+// This file works around a bug in CUDA in which the compiler miscompiles
+// atomics to automatic storage (local memory). This bug is not fixed on any
+// CUDA version yet.
+//
+// CUDA compilers < 12.3 also miscompile __isLocal, such that the library cannot
+// detect automatic storage and error. Therefore, in CUDA < 12.3 compilers this
+// uses inline PTX to bypass __isLocal.
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+_CCCL_DEVICE inline bool __cuda_is_local(const void* __ptr)
+{
+#if defined(_CCCL_CUDACC_BELOW_12_3)
+  int __tmp = 0;
+  asm("{\n\t"
+      "  .reg .pred p;\n\t"
+      "  isspacep.local p, %1;\n\t"
+      "  @p mov.s32 %0, 1;\n\t"
+      "}\n\t"
+      : "=r"(__tmp)
+      : "l"(__ptr));
+  return __tmp == 1;
+#else
+  return __isLocal(__ptr);
+#endif
+}
+
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_and(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom & __v;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_or(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom | __v;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_xor(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom ^ __v;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_add(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom + __v;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_sub(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom - __v;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_max(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __atom < __v ? __v : __atom;
+}
+template <class _Type>
+_CCCL_DEVICE void __cuda_fetch_local_bop_min(volatile _Type& __atom, _Type const& __v)
+{
+  __atom = __v < __atom ? __v : __atom;
+}
+
+template <class _Type>
+_CCCL_DEVICE bool __cuda_load_weak_if_local(const volatile _Type* __ptr, _Type* __ret)
+{
+#if defined(_LIBCUDACXX_ATOMIC_UNSAFE_AUTOMATIC_STORAGE)
+  if (!__cuda_is_local((const void*) __ptr))
+  {
+    return false;
+  }
+  memcpy((void*) __ret, (void const*) __ptr, sizeof(_Type));
+  // Required to workaround a compiler bug, see nvbug/4064730
+  __nanosleep(0);
+  return true;
+#else
+  return false;
+#endif
+}
+
+template <class _Type>
+_CCCL_DEVICE bool __cuda_store_weak_if_local(volatile _Type* __ptr, _Type __val)
+{
+#if defined(_LIBCUDACXX_ATOMIC_UNSAFE_AUTOMATIC_STORAGE)
+  if (!__cuda_is_local((const void*) __ptr))
+  {
+    return false;
+  }
+  memcpy((void*) __ptr, (void const*) &__val, sizeof(_Type));
+  return true;
+#else
+  return false;
+#endif
+}
+
+template <class _Type>
+_CCCL_DEVICE bool
+__cuda_compare_exchange_weak_if_local(volatile _Type* __ptr, _Type* __expected, const _Type* __desired, bool* __success)
+{
+  if (!__cuda_is_local((const void*) __ptr))
```

Comment:
We are missing the `#if defined(_LIBCUDACXX_ATOMIC_UNSAFE_AUTOMATIC_STORAGE)`. 
I'd expect that we'd want it here for the same reason we want it in the other ops.